Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ProblemsController.cs

```diff
@@ -821,18 +867,14 @@ private void ValidateUploadedFile(string propertyName, HttpPostedFileBase file)
             }
         }
 
-        private void RetestSubmission(int submissionId)
+        private void SetSubmissionForRetest(int submissionId)
         {
-            var submission = new Submission
-            {
-                Id = submissionId,
-                Processed = false,
-                Processing = false
-            };
-            this.Data.Context.Submissions.Attach(submission);
+            var submission = this.Data.Submissions.GetById(submissionId);
+
+            submission.Processed = false;
+
             var submissionEntry = this.Data.Context.Entry(submission);
             submissionEntry.Property(pr => pr.Processed).IsModified = true;
```

Comment:
Have you considered using batch Update from the EntityFramework.Extended library? [More info here](https://github.com/zzzprojects/EntityFramework-Plus/wiki/EF-Batch-Update-%7C-Entity-Framework-Batch-Update-Objects-Without-Loading-Them)